packages/devtools_app/lib/src/memory/flutter/memory_snapshot_models.dart

```diff
@@ -343,27 +409,73 @@ class ObjectFieldReference extends FieldReference {
           instance,
           type,
           name,
-          onExpand: (reference) {
-            // Need to construct the children.
-            if (reference.children.isNotEmpty &&
-                reference.children.first.isEmptyReference) {
-              reference.children.clear();
+        ) {
+    onExpand = _expandObjectFieldRef;
+  }
 
-              // Null value nothing to expand.
-              if (reference.isScaler) return;
+  void _expandObjectFieldRef(reference) {
+    // Need to construct the children.
+    if (reference.children.isNotEmpty &&
+        reference.children.first.isEmptyReference) {
+      // Remove empty entries compute the real values.
+      reference.children.clear();
 
-              assert(reference.isObject);
-              final objectFieldReference = reference as ObjectFieldReference;
+      // Null value nothing to expand.
+      if (reference.isScaler) return;
+    }
 
-              final objFields = instanceToFieldNodes(
-                controller,
-                objectFieldReference.instance,
-              );
+    assert(reference.isObject);
+    final ObjectFieldReference objectFieldReference = reference;
 
-              objectFieldReference.addAllChildren(objFields);
-            }
-          },
-        );
+    var objFields = instanceToFieldNodes(
+      controller,
+      objectFieldReference.instance,
+    );
+
+    if (objFields.isEmpty) {
+      final className =
+          (objectFieldReference.instance.theClass as HeapGraphClassLive).name;
+      if (className.endsWith('Array') || className.endsWith('List')) {
```

Comment:
this seems really risky. Can we move this to a helper that ideally checks whether the class implements List?

Reply:
This is old code I forgot to remove!  Now its gone.